src/components/buttonScroll.js

```diff
@@ -0,0 +1,50 @@
+import React, {  useState } from 'react'
+import { Button } from './button'
+
+export function ButtonScroll({ children, ...props }) {
+  const [showScroll, setShowScroll] = useState(false)
+
+  const scrollTop = () =>{
+    window.scrollTo({top: 0, behavior: 'smooth'})
+  };
+
+  const checkScrollTop = () => {
+    if (!showScroll && window.pageYOffset > window.innerHeight){
+      setShowScroll(true)
+    } else if (showScroll && window.pageYOffset <= window.innerHeight){
+      setShowScroll(false)
+    }
+  }
```

Comment:
É uma boa também você envolver estas funções em um hook de callback (useCallback), isto impede que qualquer atualização feita nos valores de estado acabem provocando um _re-rendering_ do componente `<ButtonScroll />`, a sugestão fica assim:
```suggestion
  const scrollTop = useCallback(() =>{
    window.scrollTo({top: 0, behavior: 'smooth'})
  }, []);

  const checkScrollTop = useCallback(() => {
    if (!showScroll && window.pageYOffset > window.innerHeight){
      setShowScroll(true)
    } else if (showScroll && window.pageYOffset <= window.innerHeight){
      setShowScroll(false)
    }
  }, [showScroll, setShowScroll])
```